yt_dlp/extractor/youmaker.py

```diff
@@ -0,0 +1,430 @@
+            'upload_date': '20211001',
+            'uploader': 'user_71885a31e113614751e14bba45d3bdcfd10d3f08',
+            'timestamp': 1633055950,
+            'subtitles': {
+                'en': [{
+                    'url': r're:https?://[a-z1-3]+.youmaker.com/assets/2021/1001/b58f88fe-4ddb-4c11-bccf-46f579b7d978/subtitle_1633055993844.auto.en.vtt'
+                }]
+            },
+        },
+        'params': {'skip_download': True, 'listsubtitles': True},
+        'expected_warnings': ['Missing m3u8 info\\. Trying alternative server'],
+    }, {
+        # all videos from channel
+        'url': 'http://www.youmaker.com/channel/f06b2e8d-219e-4069-9003-df343ac5fcf3',
+        'playlist_mincount': 30,
+        'info_dict': {
+            'id': 'f06b2e8d-219e-4069-9003-df343ac5fcf3',
+            'title': 'YoYo Cello',
+            'description': 'Connect the World Through Music. \nConnect Our Hearts with Music.',
+        },
+    }, {
+        # all videos from channel playlist
+        'url': 'https://www.youmaker.com/channel/f8d585f8-2ff7-4c3c-b1ea-a78d77640d54/playlists/f99a120c-7a5e-47b2-9235-3817d1c12662',
+        'playlist_mincount': 9,
+        'info_dict': {
+            'id': 'f99a120c-7a5e-47b2-9235-3817d1c12662',
+            'title': 'Mini Cakes',
+        },
+    }, {
+        'url': 'https://www.youmaker.com/embed/Dnnrq0lw8062/',
+        'only_matching': True,
+    }, {
+        'url': 'https://vs.youmaker.com/v/Dnnrq0lw8062/',
+        'only_matching': True,
+    }, {
+        'url': 'https://youmaker.com/playlist/v6aLJnrqkoXO/',
+        'only_matching': True,
+    }, {
+        'url': 'http://youmaker.com/channel/ntd/',
+        'only_matching': True,
+    }]
+    REQUEST_LIMIT = 50
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YoumakerIE, self).__init__(downloader=downloader)
+        self._protocol = 'https'
+        self._category_map = None
+        self._cache = {}
+
+    @staticmethod
+    def _extract_url(webpage):
+        match = re.search(
+            r"""(?x)
+                <iframe[^>]+src="
+                (?P<url>
+                    https?://(?:[a-z][a-z0-9]+\.)?
+                    youmaker\.com/embed/[0-9a-zA-Z-]+
+                )
+                [^"]*"
+                """, webpage)
+        return match.group('url') if match else None
+
+    def _fix_url(self, url):
+        if url.startswith('//'):
+            return f'{self._protocol}:{url}'
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url('//www.youmaker.com')
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url('//vs.youmaker.com/assets')
+
+    def _live_url(self, video_id):
+        return self._fix_url(f'//live.youmaker.com/{video_id}/playlist.m3u8')
```

Comment:
Unnecessary functions. These should just be inlined or normal variables. Definitely they shouldn't be `property`